src/roinput.rs

```diff
@@ -0,0 +1,658 @@
+use super::*;
+
+use algebra::{
+    PrimeField, // for size_in_bits()
+};
+
+use bitvec::{
+    prelude::*,
+    view::AsBits,
+};
+
+pub trait Input : Copy {
+    fn to_roinput(self) -> ROInput;
```

Comment:
I recommend doing three things:

- copy the github actions from the proof-systems folder so you can run them here as well (and get generated doc, formatting in CI, etc.) ([this](https://github.com/o1-labs/proof-systems/blob/master/.github/workflows/rust.yml) and [this](https://github.com/o1-labs/proof-systems/blob/master/.github/workflows/rustdoc.yml))
- add `#![warn(missing_docs)]` to the top of your `lib.rs` file, like here: https://github.com/o1-labs/proof-systems/pull/229/files
- check if `cargo doc` looks good :)

Reply:
I guess since this will be moved into a crate within proof-systems this does not need to be done.

Reply:
Docs done